src/lib/dlt_user.c

```diff
@@ -4761,6 +4844,79 @@ DltReturnValue dlt_user_log_check_user_message(void)
                         return DLT_RETURN_ERROR;
                 }
                 break;
+#ifdef DLT_TRACE_LOAD_CTRL_ENABLE
+                case DLT_USER_MESSAGE_TRACE_LOAD:
+                {
+                    /*
+                     * at least user header and message length is available
+                     */
+                    trace_load_settings_user_message_bytes_required =
+                        (int32_t) (sizeof(DltUserHeader) + sizeof(uint32_t ));
+                    if (receiver->bytesRcvd < (int32_t)trace_load_settings_user_message_bytes_required) {
+                        // Not enough data to read the message length
+                        leave_while = 1;
```

Comment:
if it is possible, kindly please using the meaning name instead of "leave_while".

Reply:
Won't do, leave_while is pr existing code. This PR introduces enough changes already so I won't be doing refactorings here. 
@minminlittleshrimp does that work for you ?

Reply:
To me, I am okay with current changes and I think it can be delivered/merged anytime.
Further refactoring and investigating can be spent on other PRs later under enhancement/refactor labels.

TBD from me to Alex in any future PR: why we dont make use of original dlt parser and using filter section concept like in dltlogstorage.conf, dlt gateway.conf but to reimplement new one just for trace limit?

Reply:
> To me, I am okay with current changes and I think it can be delivered/merged anytime.

I'll fix the remaining findings from @Bichdao021195, especially the malloc comments. 

> using filter section concept like in dltlogstorage.conf

First of all because I wasn't aware of the concept. But isn't this key value only? For trace load limits we have up to two values per key

Reply:
Hello @alexmohr 

The parser supports detecting the section and parsing the field, for the field it can be,saying:

[LIMIT1]
Apid=LOG
Ctid=TEST
Soft=
Hard=

I might think of smt like this, but yeah, just get this huge feature merged because I dont see any reasons to hold it for now. I can then raise a PR for this if I have time to do so.

I recommend to soon merge this change as it looks useful to dlt to have this feature added.

Hello @michael-methner 
Could you spend sometimes review before we merge it? Your comment will be valuable to us.

To DLT team, if you guys still need times then we can prepare some questions for Q&A after the demo section, when everything is clear and we can see how the actual working limit from the author perspective. For now I suggest we go applying this change and test under high traffic of, e.g, QNX, Android. A Rasp4 already up and run on my desk with AOSP14 so I also try this myself 😋

Reply:
Remaining findings from @Bichdao021195 have been done and the ones I don't want to change because I think it's good practice to properly init and reset values are commented. 
Looking forward to the demo next week :)